src/latest_grue_data.rs

```diff
@@ -14,6 +14,10 @@ impl LatestGrueData {
             .and_modify(|x| *x = number_of_merchandise)
             .or_insert(number_of_merchandise);
     }
+
+    pub fn get_data(&self, zone_id: u8) -> Option<&u8> {
```

Comment:
This should require a `LoadingZone` type to enforce strong type constraints.

Reply:
J'ai l'impression que c'est mieux de faire la validation à ce niveau que dans la couche UI du serveur

Reply:
Currently, your code does not validate that the number, because you assign directly to the public member, which bypass any verification in the `.try_from()` function.

I always prefer to have non-primitive types as early as possible, because it allows us to better understand what the function expects. Here, the function expecting `u8` does not tell us anything particular about this function. However `get_data(&self, zone_id: LoadingZone)` tells us that this `id` parameter *is* a loading zone id, which is between 0 and 99.

Thus, I would refactor it to make the conversion between a `u8` and `LoadingZone` as early as possible, which is in the path handler.